src/consensus/tx_verify.cpp

```diff
@@ -221,28 +221,27 @@ bool Consensus::CheckTxInputs(const CTransaction& tx, CValidationState& state, c
 
         // If prev is coinbase, check that it's matured
         if (coin.IsCoinBase() && nSpendHeight - coin.nHeight < COINBASE_MATURITY) {
-            return state.Invalid(false,
-                REJECT_INVALID, "bad-txns-premature-spend-of-coinbase",
+            return state.Invalid(ValidationInvalidReason::TX_MISSING_INPUTS, false, REJECT_INVALID, "bad-txns-premature-spend-of-coinbase",
```

Comment:
Maybe this is covered already, but why is premature coinbase using `TX_MISSING_INPUTS` as the reason? It's not really missing.

Reply:
In either case, the transaction may be valid sometime later, but isn't valid now and isn't valid in a block, so they're not very distinct.

Reply:
`TX_INPUTS_UNAVAILABLE`?

Reply:
I created a new TX_PREMATURE_COINBASE for this in the latest commit.